### DISCUSSION TITLE
How do I style a NumberInput?

### DISCUSSION THREAD

**Discussion by melohagan** [svelteuidev/svelteui#420] [Q&A]
Hey!

I was wondering if there's a way I can pass a background colour into the NumberInput?

<img width="290" alt="Screenshot 2023-07-13 at 12 41 01" src="https://github.com/svelteuidev/svelteui/assets/101575380/174197a2-675c-49fa-97d2-dd6f0e076ded">

I didn't notice any styling props on the docs page.

In this case, I would like the background to be dark

**Comment by BeeMargarida** [comment]
Hum, it should be applying the dark theme automatically. Are you using `SvelteUIProvider`?

**Comment by melohagan** [comment]
I think what I am looking for is the ability to set my own **colorScheme**. At the moment it is the only thing that changes the colour of the input.

I have tried using the [overrides](https://www.svelteui.org/theming/override) which didn't respect the background colour of the inputs.

I have tried using a [custom theme](https://www.svelteui.org/theming/create-styles) but the colours are ignored in this case as well. 

Furthermore, when I did an object comparison of `useSvelteUITheme()` when switching between the dark and light themes, the theme objects are identical. The only difference is the string field: *colorScheme*.

Which is why I think I need some way of defining my own colorScheme, and then doing something like: `themeObserver={"my-color-scheme"}`

**Reply by BeeMargarida** [reply]
I think for that you would need to create a new theme and then pass that theme name to the `themeObserver`.

```
<script lang="ts">
	import { createTheme, useSvelteUITheme, SvelteUIProvider } from '@svelteuidev/core';

        // this reuses the default theme and then overrides with the wanted properties
        const newTheme = createTheme('<your theme name>', {
            ...useSvelteUITheme(),
            // your new color scheme
            colors: {
               ...
            }
        });
</script>

<SvelteUIProvider class={newTheme} themeObserver={'<your theme name>'}>
	<slot />
</SvelteUIProvider>
```

More info about this in the `Theming` section - https://www.svelteui.org/theming/create-styles

**Reply by melohagan** [reply]
Thanks very much for the reply! I've had more success with with `css` import instead of using themes. 

It's working well! The code is here: https://github.com/melohagan/budibase-component-number-counter-input/blob/main/src/Component.svelte#L52

The only thing I just can't figure out is how to set the colour of the arrows. For example:
![Screenshot 2023-08-17 at 19 12 40](https://github.com/svelteuidev/svelteui/assets/101575380/59dd2b94-fb30-424e-b130-242aac9cf45c)

I want the colour of the arrows to match the text colour, but the color property doesn't seem to be working - I tried using it in a few different selectors. 

Are the up/down arrows SVGs? I wasn't able to figure out where they come from having a look through the svelteui codebase.
